### ISSUE TITLE
Improve formatting of XMLELEMENT()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12195]
The current formatting of `XMLELEMENT()` for very simple content is not nice:

```java
assertFormattingEquals(
    "select xmlelement(name e1, e2)",
    "select xmlelement(\n"
  + "  name e1,\n"
  + "  e2\n"
  + ")",
    select(xmlelement(e1, field(e2)))
);

assertFormattingEquals(
    "select xmlelement(name e1, 1)",
    "select xmlelement(\n"
  + "  name e1,\n"
  + "  1\n"
  + ")",
    select(xmlelement(e1, inline(1)))
);
```

It would be better if there weren't any newlines in this case, i.e. like this:

```java
assertFormattingEquals(
    "select xmlelement(name e1, e2)",
    "select xmlelement(name e1, e2)",
    select(xmlelement(e1, field(e2)))
);

assertFormattingEquals(
    "select xmlelement(name e1, 1)",
    "select xmlelement(name e1, 1)",
    select(xmlelement(e1, inline(1)))
);
```

This is especially true now that we use XML excessively for the `MULTISET` emulation, in some dialects: https://github.com/jOOQ/jOOQ/issues/12187

**Comment by lukaseder** [comment]
The result (query from #12187) does look a bit better now:

```sql
select
  1,
  xmlelement(
    name result,
    xmlagg(xmlelement(
      name record,
      xmlelement(name "v0", 2),
      xmlelement(
        name "v1",
        (
          select xmlelement(
            name result,
            xmlagg(xmlelement(
              name record,
              xmlelement(name "v0", 3),
              xmlelement(
                name "v1",
                (
                  select xmlelement(
                    name result,
                    xmlagg(xmlelement(
                      name record,
                      xmlelement(name "v0", 4),
                      xmlelement(
                        name "v1",
                        (
                          select xmlelement(
                            name result,
                            xmlagg(xmlelement(
                              name record,
                              xmlelement(name "v0", 5)
                            ))
                          )
                        )
                      )
                    ))
                  )
                )
              )
            ))
          )
        )
      )
    ))
  )
```

**Comment by lukaseder** [comment]
Implemented in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12196)
